docs/wiki/The CI environment - Publishing.md

```diff
@@ -0,0 +1,75 @@
+This section provides an overview of the principles the publishing is built upon, how it is set up, and how you can interact with it.
+
+- [Publishing overview](#publishing-overview)
+- [How it works](#how-it-works)
+    - [Example scenario](#example-scenario)
+    - [Output example](#output-example)
+
+
+<img src=".\media\CIEnvironment\publishingStep.png" alt="Publishing Step" height="500">
+
+# Publishing overview
+The publishing phase concludes each module's pipeline. If all previous tests succeed (i.e. no phase failed) and the pipeline is executed in the `main` or `master` branch, a new module version is published to all configured target locations. Currently we support the following target locations:
+
+- _[Template specs](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-specs?tabs=azure-powershell)_
+- _[Private Bicep registry](https://docs.microsoft.com/en-gb/azure/azure-resource-manager/bicep/private-module-registry)_
+- _[Azure DevOps universal packages](https://docs.microsoft.com/en-us/azure/devops/artifacts/concepts/feeds?view=azure-devops)_.
+   > Note: this is only available if using Azure DevOps pipelines.
+
+Besides the execution of a the publishing phase, there is also the possibility to set the switch `Publish prerelease module`. This switch makes it possible to publish a pre-release version in every workflow run that is not based on main or master. This can be controlled when running the module pipeline leveraging [Module pipeline inputs](./The%20CI%20environment%20-%20Pipeline%20design#module#module-pipeline-inputs).
+
+> **Note**<br>
+> The `version` used for publishing any artifact is the same for all three target locations, which reduces the maintenance effort.
+
+
+# How it works
+
+The publishing works as follows:
+
+1. The script `utilities/pipelines/resourcePublish/Get-ModulesToPublish.ps1` gets all changed module files including child modules and handles the logic of propagating the appropriate module version to be used:
+   1. The major (`x.0`) and minor (`0.x`) version are set based on the file `version.json` in the module folder.
+   2. The patch (`0.0.x`) version is calculated based on the number of commits on the `HEAD` ref (aka. git height). This will cause the patch version to never reset to 0 with major and/or minor increment, as specified for [semver](https://semver.org/).
```

Comment:
```suggestion
   1. The patch (`0.0.x`) version is calculated based on the number of commits on the `HEAD` ref (aka. git height). This will cause the patch version to never reset to 0 with major and/or minor increment, as specified for [semver](https://semver.org/).
```